### ISSUE TITLE
Question about number of reference frames

### ISSUE THREAD

**Issue by playerkk** [carrierlxk/COSNet#9]
Nice work!

For Table 2 in the paper, I was wondering if number of reference frames (N) is for inference time only or for both training and inference time. Thank you.

**Comment by carrierlxk** [comment]
Thanks. We only adopt multiple frames during the testing phase and pair-wise frames during the training phase due to the computation limitation. It is a good suggestion to take multiple frames during the training phase.

**Comment by playerkk** [comment]
Thanks for the reply.

But I didn't observe significant difference during testing time by varying the --sample_range parameter.

Also I noticed the implementation in the current code is "prediction segmentation fusion" instead of "attention summary fusion". Would it be possible to have a check? Thanks.

**Comment by carrierlxk** [comment]
I would check it after CVPR dl

**Comment by lscelory** [comment]
> Thanks for the reply.
> 
> But I didn't observe significant difference during testing time by varying the --sample_range parameter.
> 
> Also I noticed the implementation in the current code is "prediction segmentation fusion" instead of "attention summary fusion". Would it be possible to have a check? Thanks.

Hello, did you try to train this model by yourself? I am also interested in this work but I can't reappear his results. I did this for 3 weeks. If you did some training process work as well, I wish you can give me some advice about training details. Thanks in advance.

**Comment by playerkk** [comment]
> > Thanks for the reply.
> > But I didn't observe significant difference during testing time by varying the --sample_range parameter.
> > Also I noticed the implementation in the current code is "prediction segmentation fusion" instead of "attention summary fusion". Would it be possible to have a check? Thanks.
> 
> Hello, did you try to train this model by yourself? I am also interested in this work but I can't reappear his results. I did this for 3 weeks. If you did some training process work as well, I wish you can give me some advice about training details. Thanks in advance.

Hi, I only used his provided pre-trained model on DAVIS. I didn't try to train the model on my own.

**Comment by lscelory** [comment]
Ok, I got it. Thank you for reply.
I also downloaded the author's trained model, I found the test result is even better than his paper reporting.(mean J = 82.48% on DAVIS16 test set)
And as you noticed, the **Fusion Strategy** in the author's shared test code is **Prediction segmentation fusion**. I followed the description in paper and rewrote the fuse part code in Attention summary fusion way. However, I can't see result improvement as in paper. I really do not know why...[laugh cry]

**Comment by playerkk** [comment]
> Ok, I got it. Thank you for reply.
> I also downloaded the author's trained model, I found the test result is even better than his paper reporting.(mean J = 82.48% on DAVIS16 test set)
> And as you noticed, the **Fusion Strategy** in the author's shared test code is **Prediction segmentation fusion**. I followed the description in paper and rewrote the fuse part code in Attention summary fusion way. However, I can't see result improvement as in paper. I really do not know why...[laugh cry]

In terms of attention summary fusion, did you see any improvement, or not at all? Thanks.

**Comment by lscelory** [comment]
I can't see improvement using neither his segmentation fusion code or my own version attention summary code when load multi ref images. I don't know what happened.

**Comment by mingminzhen** [comment]
> Ok, I got it. Thank you for reply.
> I also downloaded the author's trained model, I found the test result is even better than his paper reporting.(mean J = 82.48% on DAVIS16 test set)
> And as you noticed, the **Fusion Strategy** in the author's shared test code is **Prediction segmentation fusion**. I followed the description in paper and rewrote the fuse part code in Attention summary fusion way. However, I can't see result improvement as in paper. I really do not know why...[laugh cry]

@lscelory How to get J=82.48%. Do you use default reference frame=2 without crf used?

**Comment by mingminzhen** [comment]
Actually, I can just get J=78.3% when I use default parameters and run the inference by 
``` 
python test_coattention_conf.py 
    --dataset davis 
    --gpus 0 '''



@lscelory Cloud you give me some hints? Thanks!

**Comment by lscelory** [comment]
@mingminzhen Hi, sorry for the late response.
I got 82.48% by just running his test_coattention_conf.py file with default set (use_crf=True). Actually according my experiment, the hyperparams sample_range wouldn't affect the final performance very much.

**Comment by Castile** [comment]
> Ok, I got it. Thank you for reply.
> I also downloaded the author's trained model, I found the test result is even better than his paper reporting.(mean J = 82.48% on DAVIS16 test set)
> And as you noticed, the **Fusion Strategy** in the author's shared test code is **Prediction segmentation fusion**. I followed the description in paper and rewrote the fuse part code in Attention summary fusion way. However, I can't see result improvement as in paper. I really do not know why...[laugh cry]

@lscelory  hi, bro. , would you send your training code and details to my e-mail：19125299@bjtu.edu.cn  ,  welcome to discuss!

**Comment by lscelory** [comment]
@Castile the author have released his training code. You can use the official code for you own training. I have run it and get the comparable result. More details to discuss are welcomed. Hoping you can have progress from the repo!

**Comment by panmyuan** [comment]
@lscelory could you tell more details about training ? could you tell you email address for further commuication. I trained the model according to author's training code, but just get  J = 72% , i don't know why ,please help me,thanks.

**Comment by lscelory** [comment]
> @lscelory could you tell more details about training ? could you tell you email address for further commuication. I trained the model according to author's training code, but just get J = 72% , i don't know why ,please help me,thanks.

Did you load the pretrained weight the author provided which is called *deeplab_davis_12_0.pth*? Based on this pretrained weight, the result is at least ~78% according to my experiment. My email address is lsc961029@163.com and we can communicate more details by emails. By the way, if you are a researcher from China, you can write your message by Chinese for our communication efficiency(laugh cry).
